evalml/tests/automl_tests/test_iterative_algorithm.py

```diff
@@ -70,6 +75,33 @@ class MockBinaryClassificationPipeline3(BinaryClassificationPipeline):
     return _method
 
 
+@pytest.fixture
```

Comment:
If we're adding this for the sake of these tests, I'm not sure how necessary this is 🤷 We're not using it to test that we're making the rounds of different pipelines like we are in the other iterative algo tests, so it could just be fine to use our `dummy_regression_pipeline_class` fixture and throw it in a list or something.